src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -0,0 +1,178 @@
+ * or {@code N} if the target is not null but matches no {@code case} label.
+ *
+ * @since 17
+ */
+@PreviewFeature(feature=PreviewFeature.Feature.SWITCH_PATTERN_MATCHING)
+public class SwitchBootstraps {
+
+    private SwitchBootstraps() {}
+
+    private static final MethodHandles.Lookup LOOKUP = MethodHandles.lookup();
+
+    private static final MethodHandle DO_SWITCH;
+
+    static {
+        try {
+            DO_SWITCH = LOOKUP.findStatic(SwitchBootstraps.class, "doSwitch",
+                                           MethodType.methodType(int.class, Object.class, int.class, Object[].class));
+        }
+        catch (ReflectiveOperationException e) {
+            throw new ExceptionInInitializerError(e);
+        }
+    }
+
+    /**
+     * Bootstrap method for linking an {@code invokedynamic} call site that
+     * implements a {@code switch} on a target of a reference type.  The static
+     * arguments are an array of case labels which must be non-null and of type
+     * {@code String} or {@code Integer} or {@code Class}.
+     * <p>
+     * The type of the returned {@code CallSite}'s method handle will have
+     * a return type of {@code int}.   It has two parameters: the first argument
+     * will be an {@code Object} instance ({@code target}) and the second
+     * will be {@code int} ({@code restart}).
+     * <p>
+     * If the {@code target} is {@code null}, then the method of the call site
+     * returns {@literal -1}.
+     * <p>
+     * the {@code target} is not {@code null}, then the method of the call site
+     * returns the index of the first element in the {@code labels} array starting from
+     * the {@code restart} index matching one of the following conditions:
+     * <ul>
+     *   <li>the element is of type {@code Class} and the target value
+     *       is a subtype of this {@code Class}; or</li>
+     *   <li>the element is of type {@code String} or {@code Integer} and
+     *       equals to the target.</li>
+     * </ul>
+     * <p>
+     * If no element in the {@code labels} array matches the target, then
+     * the method of the call site return the length of the {@code labels} array.
+     *
+     * @param lookup Represents a lookup context with the accessibility
+     *               privileges of the caller.  When used with {@code invokedynamic},
+     *               this is stacked automatically by the VM.
+     * @param invocationName unused
+     * @param invocationType The invocation type of the {@code CallSite} with two parameters,
+     *                       a reference type, an {@code int}, and {@code int} as a return type.
+     * @param labels case labels - {@code String} and {@code Integer} constants
+     *               and {@code Class} instances, in any combination
+     * @return a {@code CallSite} returning the first matching element as described above
+     *
+     * @throws NullPointerException if any argument is null
```

Comment:
```suggestion
     * @throws NullPointerException if any argument is {@code null}
```

same formatting nit for other occurrenace of "null"